woocommerce/google-listings-and-ads
# Product Block Editor: Preprocessing for adding the channel visibility

### Changes proposed in this Pull Request:

To be compatible with Product Block Editor, it also needs to add the channel visibility block.

Since product attributes and channel visibility are different groupings in terms of code and UI presentation, it would be better to separate them into different sections in the Product Block Editor as well. Therefore, before adding the block, it requires some preprocessing:

- Move and rename `Admin\Product\Attributes\AttributesBlock` class to `Admin\ProductBlocksService` class.
   - This way, when the block corresponding to `ChannelVisibilityMetaBox` is added to it afterward, the code structure should make more sense.
- Change to create a dedicated group in the Product Block Editor for this extension.
- Add a new section to this extension's group in the Product Block Editor for channel visibility setup.
- Clean up the temporary codes that prevent unmapped inputs from being added to the Product Block Editor.

### Screenshots:

#### 📷 Simple product

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/6431f6ad-2ebe-4432-b2fb-272d818c5599)

#### 📷 Variable product

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/ffdf13d7-aeef-4a7e-8acd-f51442001bd7)

#### 📷 Variation product

![image](https://github.com/woocommerce/google-listings-and-ads/assets/17420811/5b2ae69e-1f19-48e4-b8f4-f8a945aa494f)

### Detailed test instructions:

#### 📌 Prepare test environment

1. Please refer to [the test preparation in PR 2151](https://github.com/woocommerce/google-listings-and-ads/pull/2151#prepare-test-environment) with **Woo 8.4.0**.
   - Currently, it can be tested with [8.4.0-rc.1](https://github.com/woocommerce/woocommerce/releases/tag/8.4.0-rc.1)
1. `npm install`
1. `npm start`

#### 📌 Test the dedicated group for GL&A

1. Go to edit a simple product, e.g. the **Beanie with Logo** imported from WC's sample products.
1. Check if there is a **Google Listings & Ads** tab, which is that dedicated group.
1. Switch to the **Google Listings & Ads** tab.
1. Check if there are **Channel visibility** and **Product attributes** sections.
   - Currently, the Channel visibility section is empty.
   - The Product attributes section should have attribute fields that correspond to the current product's type.
1. Repeat the above test steps with variable product and variation product types.

### Changelog entry
